builtin/logical/pki/cert_util_test.go

```diff
@@ -0,0 +1,124 @@
+package pki
+
+import (
+	"fmt"
+	"testing"
+
+	"strings"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+func TestPki_FetchCertBySerial(t *testing.T) {
+	storage := &logical.InmemStorage{}
+
+	cases := map[string]struct {
+		Req    *logical.Request
+		Prefix string
+		Serial string
+	}{
+		"valid cert": {
+			&logical.Request{
+				Storage: storage,
+			},
+			"certs/",
+			"00:00:00:00:00:00:00:00",
+		},
+		"revoked cert": {
```

Comment:
https://github.com/hashicorp/vault/pull/2575/files/74965a87af47099b859552fdf2674038228a2c2e#diff-d9cca2f9a1b12f082084eaa700fff503R72 handles the cases for valid/revoked certs where the underlying path is already hyphenated so that there is no need to update the paths.